format/Tensor.fbs

```diff
@@ -38,16 +38,16 @@ table TensorDim {
 table Tensor {
   /// The type of data contained in a value cell. Currently only fixed-width
   /// value types are supported, no strings or nested types
-  type: Type;
+  type: Type (required);
 
   /// The dimensions of the tensor, optionally named
-  shape: [TensorDim];
+  shape: [TensorDim] (required);
 
   /// Non-negative byte offsets to advance one value cell along each dimension
-  strides: [long];
+  strides: [long] (required);
```

Comment:
In the C++ API for tensors, `strides` is allowed to be not specified in which case the strides are computed according to C-order / row-major striding. Do we want to do the same here or require that the strides be set always in the protocol message?

Reply:
Both are fine to me. Allowing omission is useful #for applications which use C-major arrays without strides. In Arrow C++ case, we always have strides internally, so we won’t omit strides.

Reply:
If we make it optional, should `indicesStrides` in `SparseTensorIndexCOO` also be optional?

Reply:
Yes, we should also make it optional for the API consistency.

Reply:
Ok. The problem is that the C++ code currently isn't ready for missing strides (it reads `indicesStrides` unconditionally).

Reply:
I see. I don't know the sparse C++ code well enough to fix myself at the moment